src/System.Management.Automation/namespaces/FileSystemContentStream.cs

```diff
@@ -1460,10 +1460,9 @@ private int RefillByteBuff()
                 _byteCount += _stream.Read(_byteBuff, _byteCount, (int)(lengthLeft - _stream.Position));
                 _stream.Position = _currentPosition;
             }
-            else if (_currentEncoding.Equals(Encoding.Unicode) ||
-                _currentEncoding.Equals(Encoding.BigEndianUnicode) ||
-                _currentEncoding.Equals(Encoding.UTF32) ||
-                _currentEncoding.Equals(Encoding.ASCII) ||
+            else if (_currentEncoding is UnicodeEncoding ||
+                _currentEncoding is UTF32Encoding ||
+                _currentEncoding is ASCIIEncoding ||
                 IsSingleByteCharacterSet())
```

Comment:
Is there a reason to remove the `BigEndianUnicode` handling here?

Reply:
Encoding.Unicode and Encoding.BigEndianUnicode both inherit from UnicodeEncoding so the one test handles both.